### DISCUSSION TITLE
MPICH passes -rpath without -Wl

### DISCUSSION THREAD

**Discussion by addy419** [pmodels/mpich#7153] [Q&A]
I compiled MPICH with nvhpc and it passes -rpath without -Wl. This is fine for nvhpc, but when I set -fc=gfortran to use the linker, it throws an error `gfortran: error: unrecognized command line option ‘-rpath’` because MPICH passes `-rpath /benchmarking/spack/opt/spack/linux-sles15-neoverse_v2/nvhpc-24.5/mpich-4.2.2-gjzndpqe3v2wdxedmwcyfewz76nvok4i/lib -lmpi`. I think a workaround for this is creating a config file and passing it through `-config` option, but I cannot find a syntax anywhere to create it.  Does anyone have an idea?

**Comment by hzhou** [comment]
You can try `--with-wrapper-dl-type=none` to disable the rpath options altogether. However, the missing `-Wl` seems to be a glitch from some other issues. What is your `mpifort -show`?
